_benchmark/user-guide/optimizing-benchmarks/randomizing-queries.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: default
+title: Running randomized workloads
+nav_order: 160
+parent: Optimizing benchmarks
+grand_parent: User guide
+---
+
+# Randomizing queries
+
+By default, OpenSearch Benchmark runs identical queries for multiple benchmark iterations. However, running the same queries repeatedly isn’t ideal for every test. For example, simulating real-world caching with many iterations of the same query isn’t accurate, as it results in one cache miss, followed by many hits. OpenSearch Benchmark lets you randomize queries in a configurable way. 
+
+For example, changing `"gte"` and `"lt"` in the following `nyc_taxis` operation creates distinct queries, resulting in unique cache entries:
+
+```json
+{
+    "name": "range",
+    "operation-type": "search",
+    "body": {
+    "query": {
+        "range": {
+        "total_amount": {
+            "gte": 5,
+            "lte": 15
+        }
+        }
+    }
+    }
+}
+```
+
+
+You can’t completely randomize the values, as then the cache would not get any hits. To get cache hits, the cache must sometimes see the same values. To account for the same values while randomizing, OpenSearch Benchmark generates a number `N` of value pairs for each randomized operation at the beginning of the benchmark. OpenSearch Benchmark stores these values in a saved list where each pair gets an index from `1` to `N`.
+
+Every time OpenSearch sends a query, OpenSearch Benchmark decides whether to use a pair of values from this saved list in the query. It does this a configurable fraction of the time, called repeat frequency (`rf`). If OpenSearch pair has encountered the value pair before, this might cause for a cache hit. For example, if `rf` = 0.7, the cache hit ratio could be up to 70%. This ratio could cause a hit, depending on the benchmark’s duration and cache size. 
+
+Saved value pairs are based on the `Zipf` distribution, which empirically matches usage traces for many real caches. For example, if the benchmark draws pair `i` with a probability proportional to `1 / i^alpha`, `alpha` is another parameter controlling how spread out the distribution is. Pairs with low indexes appear with a much higher frequency than those with high indexes.
+
```

Comment:
Line 37: In the second sentence, the second phrase doesn't appear to logically follow the first. Do we mean something like "`alpha` represents a parameter that controls the spread of the distribution"? In the last sentence, what is meant by "low indexes" and "high indexes"? Please clarify.

Reply:
This is an explanation of the [`Zipf` law of distribution](https://www.w3schools.com/python/numpy/numpy_random_zipf.asp). In our case, `i^alpha` would represent the `nth` common term. The sentence, as written, assumes previous knowledge of the `Zipf` law, however, @peteralfonsi, should we go a little deeper on this explanation?

Reply:
Also, @peteralfonsi, by "high indexes" and "low indexes", do we mean the order appear in the saved list relative to their terms?

Reply:
@natebower: Suggestion updated with additional details from @peteralfonsi. The `$` should appear as an equation in the final published page. I'll verify this before merging.

Reply:
@Naarcha-AWS I edited the first four sentences, but I'm still not following the last one. Is there extra text?

Reply:
@natebower: How does this sound? 

"Therefore, pairs with numerically lower indexes, or indexes closer to `1` in the saved list, appear with a higher frequency in the cache. Consequently, versus pairs with numerically higher indexes, or indexes closer to `N` on the saved list, appear with a lower frequency."

Reply:
I feel like there's some context missing here that's making the paragraph difficult to parse/comprehend. @kolchfa-aws Would you mind helping out with this one?

Reply:
How about (this needs to be formatted for math though):

OpenSearch Benchmark selects saved value pairs using the Zipf probability distribution, where the probability of selecting pair i is proportional to 1/i^α. In this formula, i represents the index of the saved value pair, and α controls how concentrated the distribution is. This distribution reflects usage patterns observed in real caches. Pairs with lower i values (closer to 1) are selected more frequently, while pairs with higher i values (closer to N) are selected less often.

Reply:
Works for me if it works for @peteralfonsi.

Reply:
@kolchfa-aws @Naarcha-AWS LGTM. Let's go with this. Thanks!

Reply:
Wait... I'm not sure `i` represents the numerical index value. I thought that `i` represents the term value.

Reply:
> Wait... I'm not sure `i` represents the numerical index value. I thought that `i` represents the term value.

Getting clarification on this before adjusting the suggestion. Thanks for your help @kolchfa-aws and @natebower!